src/main/java/daos/ProductDao.java

```diff
@@ -80,41 +79,41 @@ public void update(Product product) {
 
             String infoUpdate = String.format("Product with id = %d was updated to product with name = %d",
                     product.getId(), product.getName());
-            log.info(infoUpdate);
+            LOG.info(infoUpdate);
 
         } catch (SQLException e) {
-            log.error("Can`t update product", e);
+            LOG.error("Can`t update product", e);
         }
     }
 
     @Override
     public void delete(int id) {
-        log.trace("Deleting product...");
+        LOG.trace("Deleting product...");
         try {
             PreparedStatement preparedStatement = connection.prepareStatement(DELETE_BY_ID);
             preparedStatement.setInt(1, id);
             preparedStatement.executeUpdate();
         } catch (SQLException e) {
-            log.error("Can`t delete user by id", e);
+            LOG.error("Can`t delete user by id", e);
         }
     }
 
     @Override
-    public Optional<List<Product>> readAll() {
-        log.trace("Reading all products from DB...");
+    public List<Product> readAll() {
+        List<Product> products = new ArrayList<>();
+        LOG.trace("Reading all products from DB...");
         try {
             Statement statement = connection.createStatement();
             ResultSet resultSet = statement.executeQuery(SELECT_ALL);
-
-            List<Product> products = new ArrayList<>();
             while (resultSet.next()) {
                 products.add(Product.of(resultSet));
             }
-            Optional<List<Product>> optionalProducts = Optional.ofNullable(products);
-            return optionalProducts;
+            if(!products.isEmpty()){
+                return products;
+            }
         } catch (SQLException e) {
-            log.error("Can`t read all products", e);
+            LOG.error("Can`t read all products", e);
         }
-        return Optional.empty();
+        return null;
```

Comment:
Bad practice to return null. Better to return an empty list.
Fo instance:

if we return null the following code will fail
```
products.stream().foreach(.....)
```

In case empty it will work, just won't be any iteration in `foreach()`